src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -714,440 +719,859 @@ public GetDataSetMetaTests(TestApplicationFactory testApp) : base(testApp)
+                    }
+                }
+
+                foreach (var type in unspecifiedMetaDataTypes)
+                {
+                    switch (type)
+                    {
+                        case MetadataType.Filters:
+                            Assert.Empty(content.Filters);
+                            break;
+                        case MetadataType.Locations:
+                            Assert.Empty(content.Locations);
+                            break;
+                        case MetadataType.Indicators:
+                            Assert.Empty(content.Indicators);
+                            break;
+                        case MetadataType.TimePeriods:
+                            Assert.Empty(content.TimePeriods);
+                            break;
+                    }
+                }
+            }
+
+            [Fact]
+            public async Task EmptyList_AllowedValueError()
+            {
+                var response = await GetDataSetMeta(Guid.NewGuid(), types: []);
+
+                var validationProblem = response.AssertValidationProblem();
+
+                validationProblem.AssertHasAllowedValueError("types[0]");
+
+                var error = Assert.Single(validationProblem.Errors);
+
+                var errorDetail = GetErrorDetail(error);
+
+                Assert.NotNull(errorDetail);
+                Assert.Equal(2, errorDetail.Count);
+                Assert.Null(errorDetail["value"].GetString());
+                Assert.Equal(
+                    EnumUtil.GetEnums<MetadataType>(),
+                    errorDetail["allowed"].EnumerateArray().Select(e => EnumUtil.GetFromEnumValue<MetadataType>(e.GetString()!)));
+            }
+
+            [Theory]
+            [InlineData("filters", "filters")]
+            [InlineData("locations", "locations")]
+            [InlineData("indicators", "indicators")]
+            [InlineData("timePeriods", "timePeriods")]
+            [InlineData("invalid", "invalid")]
+            [InlineData(null, "")]
+            [InlineData(null, " ")]
+            public async Task InvalidMetadataType_AllowedValueError(string? invalidType, string metadataType)
+            {
+                var response = await GetDataSetMeta(Guid.NewGuid(), types: [metadataType]);
+
+                var validationProblem = response.AssertValidationProblem();
+
+                validationProblem.AssertHasAllowedValueError("types[0]");
+
+                var error = Assert.Single(validationProblem.Errors);
+
+                var errorDetail = GetErrorDetail(error);
+
+                Assert.NotNull(errorDetail);
+                Assert.Equal(2, errorDetail.Count);
+                Assert.Equal(invalidType, errorDetail["value"].GetString());
+                Assert.Equal(
+                    EnumUtil.GetEnums<MetadataType>(),
+                    errorDetail["allowed"].EnumerateArray().Select(e => EnumUtil.GetFromEnumValue<MetadataType>(e.GetString()!)));
```

Comment:
Formatting here could be better as the last line's really long. Could be:

```cs
Assert.Equal(
    EnumUtil.GetEnums<MetadataType>(), 
    errorDetail["allowed"]
        .EnumerateArray()
        .Select(e => EnumUtil.GetFromEnumValue<MetadataType>(e.GetString()!)));
```

Similar thing for other cases below